common/plugins/com.liferay.ide.ui/src/com/liferay/ide/ui/workspace/WorkbenchPreferencesTransfer.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.ui.workspace;
+
+import com.liferay.ide.core.util.FileUtil;
+import com.liferay.ide.ui.LiferayUIPlugin;
+
+import java.io.File;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.ui.internal.preferences.WorkbenchSettingsTransfer;
+
+/**
+ * @author Andy Wu
+ */
+@SuppressWarnings( "restriction" )
+public class WorkbenchPreferencesTransfer extends WorkbenchSettingsTransfer
+{
+
+    public WorkbenchPreferencesTransfer()
+    {
+        super();
+    }
+
+    @Override
+    public IStatus transferSettings( IPath newWorkspaceRoot )
+    {
+        IPath currentWorkspace = Platform.getLocation();
+
+        File srcDir = new File( currentWorkspace.toFile(), ".metadata/.plugins/org.eclipse.core.runtime/.settings" );
+        File destDir = new File( newWorkspaceRoot.toFile(), ".metadata/.plugins/org.eclipse.core.runtime/.settings" );
```

Comment:
It seems to be true, see here:
https://dzone.com/articles/create-new-eclipse-workspace-w

http://stackoverflow.com/a/2078547/620537

So i'm good with this implementation.

On Mon, Feb 22, 2016 at 7:16 PM, Andy Wu <notifications@github.com> wrote:

> In
> common/plugins/com.liferay.ide.ui/src/com/liferay/ide/ui/workspace/WorkbenchPreferencesTransfer.java
> <https://github.com/gamerson/liferay-ide/pull/1382#discussion_r53723061>:
>
> > +@SuppressWarnings( "restriction" )
> > +public class WorkbenchPreferencesTransfer extends WorkbenchSettingsTransfer
> > +{
> > +
> > +    public WorkbenchPreferencesTransfer()
> > +    {
> > +        super();
> > +    }
> > +
> > +    @Override
> > +    public IStatus transferSettings( IPath newWorkspaceRoot )
> > +    {
> > +        IPath currentWorkspace = Platform.getLocation();
> > +
> > +        File srcDir = new File( currentWorkspace.toFile(), ".metadata/.plugins/org.eclipse.core.runtime/.settings" );
> > +        File destDir = new File( newWorkspaceRoot.toFile(), ".metadata/.plugins/org.eclipse.core.runtime/.settings" );
>
> I tested it someway and worked well but not very sure , I read the
> http://www.vogella.com/tutorials/EclipsePreferences/article.html
> and think this is almost the right way to do .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gamerson/liferay-ide/pull/1382/files#r53723061>.
>



-- 
Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com